nomad/state/events_test.go

```diff
@@ -726,7 +726,7 @@ func TestEventsFromChanges_WithDeletion(t *testing.T) {
 	event := eventsFromChanges(nil, changes)
 	require.NotNil(t, event)
 
-	require.Len(t, event.Events, 1)
+	require.Len(t, event.Events, 2)
```

Comment:
This was testing something else (https://github.com/hashicorp/nomad/commit/5e34efadab66de20d1a64691916f10df2e8255f8), but incidentally assumed that a job deletion would not generate an event.